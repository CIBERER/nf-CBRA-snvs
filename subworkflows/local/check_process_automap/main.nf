include { AUTOMAP } from '../../../modules/local/automap/main'
include { CREATE_AUTOMAP_LOG } from '../../../modules/local/create_automap_log/main'

workflow CHECK_PROCESS_AUTOMAP {

    take:
    ch_vcfs        // channel (mandatory): tuple val(meta), path(vcfs), path(tbis)
    automap_assembly // channel (mandatory): val(assembly)        

    main:

    // Run AUTOMAP - failures will be ignored
    AUTOMAP (
        ch_vcfs,
        automap_assembly
    )
    
    // Create a channel with all samples and their status
    // Join input with output to determine success/failure
    status_channel = ch_vcfs
        .map { meta, vcf -> [meta, vcf] }  // Simplify to just meta and vcf
        .join(AUTOMAP.out.roh_automap_file, remainder: true)
        .map { meta, vcf, automap_file -> 
            def status = automap_file ? "SUCCESS" : "FAILED"
            [meta, status, vcf]
        }

    // Create log file
    CREATE_AUTOMAP_LOG (
        status_channel.collect()
    )

    // Emit only successful results
    successful_samples = AUTOMAP.out.roh_automap_file

    emit:
    automap = successful_samples
    log_file = CREATE_AUTOMAP_LOG.out.log_file
}
