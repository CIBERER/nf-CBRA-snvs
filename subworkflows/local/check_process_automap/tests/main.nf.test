nextflow_workflow {

    name "Test Subworkflow CHECK_PROCESS_AUTOMAP"
    script "../main.nf"
    workflow "CHECK_PROCESS_AUTOMAP"

    tag "subworkflows"
    tag "subworkflows/check_process_automap"


    test("automap - vcf") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test'], // meta map
                    file("https://raw.githubusercontent.com/mquinodo/AutoMap/refs/heads/master/Test/TestSample.vcf", checkIfExists: true)
                ])
                input[1] = "hg38"
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("automap not working - vcf") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test'], // meta map
                    file("https://zenodo.org/records/15355673/files/Sample1_T1.dragen.PASS.vcf.gz?download=1", checkIfExists: true)
                ])
                input[1] = "hg38"
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}