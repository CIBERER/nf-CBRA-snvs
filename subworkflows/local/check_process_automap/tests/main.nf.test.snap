{
    "automap - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "TestSample.HomRegions.tsv:md5,45a2378a6f760cac0e05fd1264f995bb"
                    ]
                ],
                "automap": [
                    [
                        {
                            "id": "test"
                        },
                        "TestSample.HomRegions.tsv:md5,45a2378a6f760cac0e05fd1264f995bb"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-06-23T12:43:30.433165576"
    },
    "automap not working - vcf": {
        "content": [
            {
                "0": [
                    
                ],
                "automap": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-06-23T12:43:38.19488595"
    }
}